### PR TITLE
Scheduler a11y: Appointment collector button doesn't have info about date

### PR DESCRIPTION
**Pull request by tongsonbarbs**

### CODE REVIEW

File: packages/devextreme/js/__internal/scheduler/m_compact_appointments_helper.ts [hunk 1]
```diff
@@ -181,4 +184,29 @@ export class CompactAppointmentsHelper {
...
+  _getEndDate(appointment) {
+    const date = appointment.endDate;
+    return date ? new Date(date) : null;
+  }
+
+  _getDateText(appointment) {
+    const startDateText = this._localizeDate(this._getStartDate(appointment));
+    const endDateText = this._localizeDate(this._getEndDate(appointment));
+
+    const dateText = startDateText === endDateText
```
Comment (siarhei-arzamasau) [inline packages/devextreme/js/__internal/scheduler/m_compact_appointments_helper.ts:208]: @tongsonbarbs what would happen if both of the dates would be `null`?

File: packages/devextreme/js/__internal/scheduler/m_compact_appointments_helper.ts [hunk 2]
```diff
@@ -181,4 +184,29 @@ export class CompactAppointmentsHelper {
       .append($('<span>').text(text))
       .addClass(APPOINTMENT_COLLECTOR_CONTENT_CLASS);
   }
+
+  _localizeDate(date) {
+    return `${dateLocalization.format(date, 'monthAndDay')}, ${dateLocalization.format(date, 'year')}`;
+  }
+
+  _getStartDate(appointment) {
```
Comment (siarhei-arzamasau) [inline packages/devextreme/js/__internal/scheduler/m_compact_appointments_helper.ts:192]: @tongsonbarbs what do you think if you merge `_getStartDate` and `_getEndDate` into one function which receives `appointment.date`, looks like both of these functions are doing the same job.

Comment (tongsonbarbs) [reply packages/devextreme/js/__internal/scheduler/m_compact_appointments_helper.ts:192]: I wanted to set a practice where we follow Single Responsibility Pattern.. These both functions may look similar but they return different and specific data.

File: packages/devextreme/js/__internal/scheduler/m_compact_appointments_helper.ts (hunk 1 above)
Comment (tongsonbarbs) [reply packages/devextreme/js/__internal/scheduler/m_compact_appointments_helper.ts:208]: Thanks for noticing this, I tested it and it is returning undefined as string.. I have committed the changes now to fix this..

**Review by siarhei-arzamasau — APPROVED** [review]

**Pull request merged**